### PR TITLE
#1030:add docker check in setup and close terminal after installation

### PR DESCRIPTION
**Pull request by Ttring**
- Instead of refreshing the environment variable, it closes the terminal after installation.

### CODE REVIEW

File: scripts/src/main/resources/scripts/command/docker [hunk 1]
```diff
@@ -38,55 +38,51 @@ function doResolveDownloadUrl() {
 function doSetup() {  
   if [ "${DOCKER_EDITION}" != "DOCKERDESKTOP" ]
   then
-    echo "Rancher Desktop set as option"
     if uname -r | grep -i microsoft
     then 
       doFail "It seems that you are running the setup from inside a WSL shell.\nPlease use git-bash for that purpose. Otherwise your OS will be interpreted as Linux"
     else
-      if doIsRancherInstalled
+      if doIsDockerInstalled
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/docker:45]: actually this check should be done independent of `${DOCKER_EDITION}`.
If `docker` is already present, `doSetup` has nothing left to do.

File: scripts/src/main/resources/scripts/command/docker [hunk 2]
```diff
@@ -38,55 +38,51 @@ function doResolveDownloadUrl() {
...
-          #check if vpnkit is already installed
-          doEnsureVpnKitInstallation
-        elif doIsMacOs
-        then
-          # shellcheck disable=SC2016
-          doRunCommand 'cp -a "${rancher_install_dir}"/*.app /Applications/'
         else
-          doRunCommand "cp -a \"${rancher_install_dir}/.\" \"${DEVON_IDE_HOME}/software/docker/\""
-          doRunCommand "${DEVON_IDE_HOME}/software/docker/rancher-desktop"
+          if [ "${1}" != "silent" ] && ! doIsQuiet
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/docker:61]: looks quite duplicated now...

File: scripts/src/main/resources/scripts/command/docker [hunk 3]
```diff
@@ -95,7 +91,7 @@ function doSetup() {
       if doIsWindows
       then
         doEchoStep "Installing Docker Desktop for Windows..."
-        doRunCommand "${DEVON_DOWNLOAD_DIR}/docker-stable-win.exe"
+        doRunCommand "${DEVON_DOWNLOAD_DIR}/docker-stable-windows.exe"
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/docker:94]: nice catch. I have just merged the same fix with #1037 already.

File: scripts/src/main/resources/scripts/command/docker [hunk 4]
```diff
@@ -105,11 +101,44 @@ function doSetup() {
         doRunCommand "cp -a '${mount_dir}'/*.app /Applications/"
         doRunCommand "hdiutil detach -force '${mount_dir}'"
       fi
-      read -r -p "Docker Desktop has been installed. Please read https://github.com/devonfw/ide/blob/master/documentation/docker.asciidoc#setup for enabling kubernetes and further important details. Press enter to continue" answer
+      read -r -p "Docker Desktop has been installed. Please read https://github.com/devonfw/ide/blob/master/documentation/docker.asciidoc#setup for enabling kubernetes and further important details. Press enter to continue" answer     
+      kill -9 $PPID
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/docker:105]: kill -9 $PPID will kill the parent process what is the shell.
This looks kind of dangerous and hackish. I am not sure if this is the way to go...

**Review by hohwille — CHANGES_REQUESTED** [review]
@Ttring thanks for your PR and efforts to improve devonfw-ide. 👍 
As some changes can not be merged as is, I would kindly like to ask how to proceed.
Will you have capacity to rework? Or should we incorporate some of your ideas and changes ourselves?

**Comment by Amueller36** [comment]
https://github.com/devonfw/ide/pull/1045 
I did a pull request regarding this issue

**Pull request closed without merging**